app/src/main/java/com/artemzin/qualitymatters/ApplicationModule.java

```diff
@@ -33,6 +34,11 @@ public QualityMattersApp provideQualityMattersApp() {
     }
 
     @Provides @NonNull @Singleton
+    public Context provideContext() {
```

Comment:
Yes, storing a static reference to an Activity context is to be avoided. Storing a non-static reference to the Application context (which the original method returned) is safe. Just sayin' :) But no problem this way either.